x-pack/protocol/src/main/java/org/elasticsearch/protocol/xpack/graph/GraphExploreResponse.java

```diff
@@ -198,5 +216,59 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         return builder;
     }
 
+    public static GraphExploreResponse fromXContext(XContentParser parser) throws IOException {
+        GraphExploreResponse result = new GraphExploreResponse();  
+        result.vertices = new HashMap<>();
+        result.connections = new HashMap<>();
+        ensureExpectedToken(Token.START_OBJECT, parser.nextToken(), parser::getTokenLocation);
+        parser.nextToken();
+        String currentFieldName = parser.currentName();
+        List<Vertex> vertices = new ArrayList<>();
+        List<ShardSearchFailure> failures = new ArrayList<>();
+        List<Connection.UnresolvedConnection> unresolvedConnections = new ArrayList<>();
+        for (Token token = parser.nextToken(); token != Token.END_OBJECT; token = parser.nextToken()) {
```

Comment:
This one also looks right when I scan it but'd be easier to review as `ObjectParser`.

Reply:
I may be missing something but isn't ObjectParser more geared towards parsing "request" objects? Requests tend to have existing "Builder" classes with setter methods used by clients to express criteria and ObjectParser relies on registering setter methods.

Response objects tend to be immutable data objects and adding "setter" methods to accommodate the use of ObjectParser is therefore perhaps the wrong approach?

Reply:
`ConstructingObjectParser` can mostly get the job done without needing any extra builders. But sometimes builders are easier to understand. The idea with using these parsers is that they are a ton easier to review for correctness, especially that they properly ignore new fields. It isn't required, but it does make it easier to be sure that it is all rigged up right.